pkg/ocm/ocm.go

```diff
@@ -237,9 +239,57 @@ func (c *SdkClient) LimitedSupportExists(ls LimitedSupportReason, internalCluste
 	return false, nil
 }
 
+// isLimitedSupportReasonFlapping checks if a cluster has been put in limited support too many times
+// Returns true if the cluster has been put in limited support at least 2 times in the last 4 hours
+func (c *SdkClient) isLimitedSupportReasonFlapping(ls LimitedSupportReason, internalClusterID string) (bool, error) {
+	// Retrieve external ID
+	info, err := c.GetClusterInfo(internalClusterID)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve cluster info: %w", err)
+	}
+
+	externalID := info.ExternalID()
+
+	// Rerieve service logs
+	resp, err := c.conn.ServiceLogs().V1().Clusters().Cluster(externalID).ClusterLogs().List().Send()
+	if err != nil {
+		return false, fmt.Errorf("could not list service logs history: %w", err)
+	}
+
+	serviceLogs, ok := resp.GetItems()
+	if !ok {
+		return false, nil
+	}
+
+	// Check if the cluster has been put in limited support at least 2 times in the last 4 hours
+	fourHoursAgo := time.Now().Add(-4 * time.Hour)
+
+	var filteredLogs []*servicelogsv1.LogEntry
+	for _, log := range serviceLogs.Slice() {
+		if log.ServiceName() == "LimitedSupport" &&
+			log.Username() == "service-account-ocm-cad-production" &&
```

Comment:
One more small question: should we add staging here as well, so that it behaves the same on staging and prod for testing?

Reply:
I also added the staging account to the condition.